ms/code/OrdersUI.java

```diff
@@ -190,6 +192,32 @@ public static void main(String args[])
 
 			} // if
 
+			//////////// option 4 ////////////
+
+			if ( option == '4' )
```

Comment:
I think there should be a while loop. Pls refer to option 2 (retrieveWithOrderId)